### ISSUE TITLE
Use cases for this action over official Vercel git integration?

### ISSUE THREAD

**Issue by leerob** [amondnet/vercel-action#103]
Hello! I'd love to hear more about what this action aims to solve that the official Vercel git integration does not.

- What is missing from the official git integration?
- Why did you choose this over the default solution?
- What could we do better?

We'd had some confusion since this is **not a official Vercel solution** and I'd love to have some clarity. Thank you!

https://vercel.com/docs/git

**Comment by vadym-vorobel** [comment]
Hi @leerob! I just want to share my use case for this action with you.

We have our NodeJS application in a private GitHub repository. Also, it is using the NPM package with another private GitHub repository as a source.

To install this package I need an SSH key to be present in the SSH agent. Right now it's not possible with Vercel itself. See more in this issue - https://github.com/vercel/vercel/discussions/5274

So, we've decided to use GitHub Actions + this vercel-action to build the application correctly and publish it to Vercel.

I hope, it makes sense.

**Comment by DLevai94** [comment]
@leerob I second @dszymczuk 

Also, I only need 1 member of the Vercel team but this way none of the PR's made by my GH team members are deployed. I don't want to add them just for the reason of deployment. We're a team of 4. That'd be $80 / month to let them deploy.

**Comment by euaaaio** [comment]
> * Why did you choose this over the default solution?

Clarity and consistency. When everything related to CI is in one area, has its own structure and is managed by one technology. And not mashed into different interfaces that don't depend on you, work somehow differently, and can change to the worse at any moment.

**Comment by brandonarbini** [comment]
I'm using it to get around the monorepo limit that Vercel has. I have actions that run on changes to each deployable app in Github. Just like that...no more arbitrary limit. Thank you!

**Comment by leo** [comment]
@brandonarbini Thanks for sharing! Would you mind elaborating on which limit you mean exactly?

**Comment by dszymczuk** [comment]
@leo It means that if have monorepo (fe. 3 projects in one folder), you can't initialize 3 independent projects in each folder. You can initialize vercel project only on the root folder.
Check my issue: https://github.com/vercel/vercel/discussions/5294

**Comment by leo** [comment]
@dszymczuk That's an interesting issue – thanks for sharing it! But this issue right here is about why the GitHub Action would be used instead of our [Git Integration](https://vercel.com/docs/git), not about development.

Perhaps @brandonarbini is talking about how every Git repository on Vercel can only be connected to a fixed number of projects within a single Vercel Personal Account or Team?

**Comment by brandonarbini** [comment]
Yep, there are some limitations that I need to get around like the max of 3 projects for personal accounts and 10 for teams. I prefer to work in a monorepo for my personal projects and I have more than 3 there. Also, I just saw the "ignored build step" feature that wasn't there when I was getting things running initially. That will help a lot because I don't like building everything on every push. Ultimately, the limit of 3 on personal is limiting and upgrading to a team plan for $20/mo seems unnecessary since I only deploy changes to these projects very occasionally.

**Comment by leo** [comment]
I see. That makes a lot of sense, thank you! I'll bring this up to the team too.

**Comment by daiyi** [comment]
@leerob Hi there! My setup was too far off the platonic project shape and I was unable to get the build to work with the official git integration to work. 

- **lack of support for private submodules**. The biggest blocker is that I could not find a nice way to get the official integration to authenticate with the dozens of private submodules in my project. I found [this hack](https://github.com/beeinger/vercel-private-submodule) but I didn't like the complexity of adding a shell script. 
- I need to access the preview url for the build step of my project. I couldn't figure out how to do this in Vercel's native GitHub integration. I couldn't figure out how to do this with GitHub actions either, but I was able to find a workaround where I assign an alias after deploying.
- members of my GitHub org can transparently see and modify the build process without needing access to Vercel. I work on a project with rotating contractors coming and going frequently. It's a logistical pain to have to add and remove people to a vercel team who want to even look at the build. It's much simpler to move the build into GitHub Actions, and grant permissions one time through GitHub

**Comment by gnovotny** [comment]
Hi @leerob. Thanks for asking!

My understanding is that companies make decisions based on profits, and making it easier for non-team members to occasionally trigger a deployment is not financially viable for Vercel as a company. 

Unfortunately, forcing all my post-handover clients to pay $20 each for extra team members just so we can occasionally push a hotfix isn't financially viable for our own companies. 

Vercel CLI is still a thing of course, but the intentional roadblocks and phasing out of legacy features (legacy domain aliases stopped working for me recently) forced my hand. 

I hope you understand. Thanks for all the great work on Next.js!

**Comment by graup** [comment]
Hi @leerob 

My use case is to deploy the production branch only when there's a tag in git.
